lacommunaute/forum/models.py

```diff
@@ -86,3 +88,36 @@ def get_stats(self, days_back):
             stats_list[1].reverse()
 
         return stats
+
+    @cached_property
+    def count_engaged_users(self):
+        forums = self.get_family()
+
+        posters = User.objects.filter(posts__topic__forum__in=forums).values_list("email").distinct()
+        likers = User.objects.filter(topic_likes__forum__in=forums).values_list("email").distinct()
+        voters = (
+            User.objects.filter(poll_votes__poll_option__poll__topic__forum__in=forums).values_list("email").distinct()
+        )
+        authenticated_users = posters | likers | voters
+
+        anonymous_posters = (
+            Post.objects.filter(topic__forum__in=forums)
+            .exclude(username__isnull=True)
+            .annotate(email=F("username"))
+            .values_list("email")
+            .distinct()
+        )
+
+        return {
+            "posters": posters.count(),
+            "likers": likers.count(),
+            "voters": voters.count(),
+            "authenticated_users": authenticated_users.count(),
+            "anonymous_posters": anonymous_posters.count(),
+            "all_users": len(list(set(authenticated_users) | set(anonymous_posters))),
```

Comment:
Me semble que `len()` fonctionne sur les `set()` ;)
```suggestion
            "all_users": len(set(authenticated_users) | set(anonymous_posters)),
```